src/overlays/actors/ovl_Object_Kankyo/z_object_kankyo.c

```diff
@@ -32,41 +57,879 @@ const ActorInit Object_Kankyo_InitVars = {
+        this->demoObjectLoaded = true;
+        this->effects[0].alpha = 0;
+        this->actor.objBankIndex = this->demoObjectIndex;
+        this->effects[0].size = 7.0f;
+        ObjectKankyo_SetupAction(this, ObjectKankyo_SunGrave);
+    }
+}
+
+void ObjectKankyo_SunGrave(ObjectKankyo* this, GlobalContext* globalCtx) {
+    if (globalCtx->csCtx.state != 0) {
+        if (globalCtx->csCtx.npcActions[1] != NULL && globalCtx->csCtx.npcActions[1]->action == 2) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_BIRI_SPARK - SFX_FLAG);
+            if ((s16)this->effects[0].alpha + 20 > 255) {
+                this->effects[0].alpha = 255;
+            } else {
+                this->effects[0].alpha += 20;
+            }
+            Math_SmoothStepToF(&this->effects[0].size, 1.8f, 0.5f, 0.28f, 0.01f);
+        }
+    }
+}
+
+void ObjectKankyo_DrawSunGrave(ObjectKankyo* this2, GlobalContext* globalCtx2) {
+    ObjectKankyo* this = this2;
+    GlobalContext* globalCtx = globalCtx2;
+    Vec3f start;
+    Vec3f end;
+    f32 interp;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_object_kankyo.c", 1324);
+    if (globalCtx->csCtx.state != 0) {
+        if (globalCtx->csCtx.npcActions[1] != NULL && globalCtx->csCtx.npcActions[1]->action == 2 && this->demoObjectLoaded) {
+            // apparently, light waves with larger amplitudes look brighter, so the name 'amplitude' kind of works here
+            if (this->effects[0].state == 0) {
+                this->effects[0].amplitude += 1.0f / 7.0f;
+                if (this->effects[0].amplitude >= 1.0f) {
+                    this->effects[0].amplitude = 1.0f;
+                    this->effects[0].state++;
+                }
+            } else {
+                this->effects[0].amplitude -= 1.0f / 7.0f;
+                if (this->effects[0].amplitude <= 0.1f) {
+                    this->effects[0].amplitude = 0.0f;
+                    this->effects[0].state = 0;
+                }
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Object_Kankyo/ObjectKankyo_Destroy.s")
+            if (++this->effects[0].timer > 7) {
+                this->effects[0].timer = 0;
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Object_Kankyo/func_80BA2DC8.s")
+            start.x = globalCtx->csCtx.npcActions[1]->startPos.x;
+            start.y = globalCtx->csCtx.npcActions[1]->startPos.y;
+            start.z = globalCtx->csCtx.npcActions[1]->startPos.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Object_Kankyo/func_80BA2DD4.s")
+            end.x = globalCtx->csCtx.npcActions[1]->endPos.x;
+            end.y = globalCtx->csCtx.npcActions[1]->endPos.y;
+            end.z = globalCtx->csCtx.npcActions[1]->endPos.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Object_Kankyo/ObjectKankyo_Update.s")
+            interp = func_8006F93C(globalCtx->csCtx.npcActions[1]->endFrame,
```

Comment:
```suggestion
            weight = func_8006F93C(globalCtx->csCtx.npcActions[1]->endFrame,
```
see [`Kankyo_LerpWeight`](https://github.com/petrie911/oot/blob/Kankyo/src/code/z_kankyo.c#L482) (`func_8006F93C`)